src/Controller/ApiController.php

```diff
@@ -48,21 +46,21 @@ function generateList($request, $products): \Generator
     }
 
     /**
-     * @param Request $request
-     * @param string  $_sku
+     * @param string              $_sku
+     * @param TranslatorInterface $translator
      *
      * @return JsonResponse
      */
     #[Route('/api/product{_sku}', name: 'product_info')]
-    public function getProductInfo(Request $request, string $_sku): JsonResponse
+    public function getProductInfo(string $_sku, TranslatorInterface $translator): JsonResponse
     {
-        $this->isRequestContentTypeJson($request);
-
         $currentProduct = [];
         $this->getAllProducts();
         foreach ($this->products as $product) {
             if (mb_substr($product['sku'], -4) === $_sku) {
-                $currentProduct = $product;
+                array_walk($product, function ($item, $key) use (&$currentProduct, $translator) {
```

Comment:
Not the cleanest solution. As there are not many fields in a product, it's more straightforward to print them one by one.